### PR TITLE
Fix/issuer-integration

### PR DESCRIPTION
**Pull request by Kolezhniuk**
JS SDK issuer fixes

### CODE REVIEW

File: src/iden3comm/handlers/credential-proposal.ts [hunk 1]
```diff
@@ -218,16 +219,37 @@ export class CredentialProposalHandler
...
       // check if there is credentials in the wallet
       let credsFromWallet: W3CCredential[] = [];
+
       try {
-        credsFromWallet = await this._identityWallet.findOwnedCredentialsByDID(senderDID, {
-          type: cred.type,
-          context: cred.context
-        });
+        if (isExternalSender) {
+          credsFromWallet = await this._credWallet.findByQuery({
```
Comment (vmidyllic) [inline src/iden3comm/handlers/credential-proposal.ts:238]: why can't we always use this? if it is 'internal' sender I think it work in the same way.

File: src/iden3comm/handlers/message-handler.ts [hunk 2]
```diff
@@ -3,6 +3,10 @@ import { AuthMessageHandlerOptions } from './auth';
 import { RevocationStatusMessageHandlerOptions } from './revocation-status';
 import { ContractMessageHandlerOptions } from './contract-request';
 import { PaymentHandlerOptions, PaymentRequestMessageHandlerOptions } from './payment';
+import { MediaType } from '../constants';
+import { proving } from '@iden3/js-jwz';
+import { DID } from '@iden3/js-iden3-core';
+import { PROTOCOL_CONSTANTS } from '..';
```
Comment (vmidyllic) [inline src/iden3comm/handlers/message-handler.ts:9]: is '..' a right import?

File: src/iden3comm/handlers/message-handler.ts [hunk 3]
```diff
@@ -120,7 +124,7 @@ export class MessageHandler {
       | RevocationStatusMessageHandlerOptions
       | PaymentRequestMessageHandlerOptions
       | PaymentHandlerOptions
-      | { [key: string]: unknown }
+      | { senderDID?: DID; [key: string]: unknown }
```
Comment (vmidyllic) [inline src/iden3comm/handlers/message-handler.ts:127]: should we use senderDID or senderDid

File: src/iden3comm/handlers/credential-proposal.ts [hunk 4]
```diff
@@ -165,6 +165,7 @@ export class CredentialProposalHandler
   constructor(
     private readonly _packerMgr: IPackageManager,
     private readonly _identityWallet: IIdentityWallet,
+    private readonly _credWallet: ICredentialWallet,
```
Comment (vmidyllic) [inline src/iden3comm/handlers/credential-proposal.ts:168]: looks like breaking change, can't we make getter for credential wallet in identity wallet instance? or create function in identity wallet that will proxy call for query?

File: src/iden3comm/constants.ts [hunk 5]
```diff
@@ -72,3 +72,6 @@ export const SUPPORTED_PUBLIC_KEY_TYPES = {
     'JsonWebKey2020'
   ]
 };
+
+export const DEFAULT_PROOF_VERIFY_OPT = 1 * 60 * 60 * 1000; // 1 hour
```
Comment (volodymyr-basiuk) [inline src/iden3comm/constants.ts:76]: maybe specify in name that it's delay ?
Something like `DEFAULT_PROOF_VERIFY_DELAY_OPT`
And `DEFAULT_AUTH_VERIFY_DELAY_OPT` (or OPTS in both at the end)

File: src/iden3comm/handlers/message-handler.ts [hunk 6]
```diff
@@ -131,10 +134,20 @@ export class MessageHandler {
...
-    if (response) {
-      return this._params.packageManager.packMessage(unpackedMediaType, response, {});
+    if (!response) {
+      return null;
     }
 
-    return null;
+    let packerParams = {};
+    const senderDID = (context as { senderDID?: DID })?.senderDID;
+    if (unpackedMediaType === MediaType.ZKPMessage && senderDID) {
```
Comment (volodymyr-basiuk) [inline src/iden3comm/handlers/message-handler.ts:143]: Shouldn't it throw if `unpackedMediaType` is `MediaType.ZKPMessage` but no `senderDID` in context ?

File: src/proof/proof-service.ts [hunk 7]
```diff
@@ -494,7 +517,19 @@ export class ProofService implements IProofService {
...
+        getDateFromUnixTimestamp(Number(globalStateInfo.replacedAtTimestamp)).getTime();
+
+      if (
+        timeDiff >
+        (opts?.acceptedStateTransitionDelay ?? PROTOCOL_CONSTANTS.DEFAULT_AUTH_VERIFY_OPTS)
+      ) {
+        throw new Error('global state is outdated');
+      }
+
+      return true;
```
Comment (volodymyr-basiuk) [inline src/proof/proof-service.ts:532]: let's remove this return true
Generaly looks like this function throws or return true, could be `void`, but for now I guess it's ok to have just one `return true` at the end

**Review by volodymyr-basiuk — APPROVED** [review]

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**